pkg/adapter/awssqs/adapter.go

```diff
@@ -0,0 +1,188 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awssqs
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/knative/pkg/cloudevents"
+	"github.com/knative/pkg/logging"
+	"github.com/knative/pkg/signals"
+	"go.uber.org/zap"
+	"golang.org/x/net/context"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+const (
+	eventType = "aws.sqs.message"
+)
+
+// Adapter implements the AWS SQS adapter to deliver SQS messages from a
+// an SQS queue to a Sink.
+type Adapter struct {
+
+	// Region is the AWS region
+	Region string
+
+	// QueueUrl is the AWS SQS URL that we're polling messages from
+	QueueUrl string
+
+	// SinkURI is the URI messages will be forwarded on to.
+	SinkURI string
+
+	// CredsFile is the full path of the AWS credentials file
+	CredsFile string
+}
+
+func (a *Adapter) Start(ctx context.Context) error {
+
+	logger := logging.FromContext(ctx)
+
+	stopCh := signals.SetupSignalHandler()
```

Comment:
What is the purpose in this stop channel? Normally they get passed in, so something on the outside can stop this function.

Reply:
I copied this from @rootfs 's first attempt.  My understanding is that the SetupSignalHandler returns a channel that will emit on SIGTERM and SIGINT (see https://github.com/kubernetes/apiserver/blob/master/pkg/server/signal.go)

I'll leave it for now but let me know if that's incorrect (or the stop should be implemented in other ways later).

Reply:
Sorry, my head must have been in `eventing` when I wrote this comment, where stop channels are almost always passed in. 

I still have a preference for the stop channel to be defined in `main()` and then passing it in, but that doesn't appear to be a pattern in `eventing-sources` yet, so I won't force it.